table/server/common/src/main/java/alluxio/table/common/udb/UdbFilterSpec.java

```diff
@@ -0,0 +1,505 @@
+        return false;
+      }
+      PatternNameTemplate that = (PatternNameTemplate) o;
+      return Objects.equals(mPattern, that.mPattern);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(mPattern);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this)
+          .add("pattern", mPattern)
+          .toString();
+    }
+  }
+
+  /**
+   * Builder for {@link UdbFilterSpec}.
+   */
+  public static class Builder {
+    private Set<NameTemplate> mBypassedTables;
+    private Set<NameTemplate> mIgnoredTables;
+    private Map<String, Set<NameTemplate>> mBypassedPartitions;
+    private Map<String, Mode> mPartitionBypassModes;
+    private Mode mTableBypassMode;
+    private Mode mIgnoreMode;
+
+    /**
+     * Creates a new builder.
+     */
+    public Builder() {
+      mTableBypassMode = Mode.NONE;
+      mIgnoreMode = Mode.NONE;
+      mBypassedTables = new HashSet<>();
+      mIgnoredTables = new HashSet<>();
+      mBypassedPartitions = new HashMap<>();
+      mPartitionBypassModes = new HashMap<>();
+    }
+
+    /**
+     * Include or exclude the bypassed tables.
+     * @param mode {@link Mode} of the tables
+     * @return this builder
+     */
+    public Builder setBypassedTablesMode(Mode mode) {
+      mTableBypassMode = mode;
+      return this;
+    }
+
+    /**
+     * Include or exclude the ignored tables.
+     * @param mode {@link Mode} of the tables
+     * @return this builder
+     */
+    public Builder setIgnoredTablesMode(Mode mode) {
+      mIgnoreMode = mode;
+      return this;
+    }
+
+    /**
+     * Add a table by its exact name to the bypass entry.
+     * @param name table name
+     * @return this builder
+     */
+    public Builder addBypassedTable(String name) {
+      return addTable(new ExactNameTemplate(name), Entry.BYPASS);
+    }
+
+    /**
+     * Add a table by pattern to the bypass entry.
+     * @param pattern pattern
+     * @return this builder
+     */
+    public Builder addBypassedTable(Pattern pattern) {
+      return addTable(new PatternNameTemplate(pattern), Entry.BYPASS);
+    }
+
+    /**
+     * Add a table by its exact name to the ignore entry.
+     * @param name table name
+     * @return this builder
+     */
+    public Builder addIgnoredTable(String name) {
+      return addTable(new ExactNameTemplate(name), Entry.IGNORE);
+    }
+
+    /**
+     * Add a table by pattern to the ignore entry.
+     * @param pattern pattern
+     * @return this builder
+     */
+    public Builder addIgnoredTable(Pattern pattern) {
+      return addTable(new PatternNameTemplate(pattern), Entry.IGNORE);
+    }
+
+    private Builder addTable(NameTemplate item, Entry entry) {
+      switch (entry) {
+        case BYPASS:
+          mBypassedTables.add(item);
+          break;
+        case IGNORE:
+          mIgnoredTables.add(item);
+          break;
+        case NONE:
+        default:
+          // unreachable
```

Comment:
if this is really unreachable, let's throw an IllegalStateException exception or at least `LOG.error`

Reply:
Fixed in 2719fbf31d84e0f3f486d1892a734da5f3471e62

Reply:
Throws IllegalStateException in 2719fbf31d84e0f3f486d1892a734da5f3471e62